src/crypto/CHIPCryptoPALPSA.cpp

```diff
@@ -675,7 +689,7 @@ CHIP_ERROR P256Keypair::Initialize(ECPKeyTarget key_target)
     else if (key_target == ECPKeyTarget::ECDSA)
     {
         psa_set_key_algorithm(&attributes, PSA_ALG_ECDSA(PSA_ALG_SHA_256));
-        psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_EXPORT | PSA_KEY_USAGE_SIGN_MESSAGE);
+        psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_SIGN_MESSAGE | PSA_KEY_USAGE_COPY);
```

Comment:
The problem with `PSA_KEY_USAGE_COPY` is that it is not guaranteed to work on all platforms. E.g. some crypto hardware may not support copying keys. That's why I implemented `PSAOperationalKeystore` using non-volatile pending keys to make it more portable. I don't know how to solve that but maybe we just need two implementations of `PSAOperationalKeystore` (in which case maybe you could keep and rename the existing implementation to sth like `PSAOperationalKeystoreNocopy`).

Reply:
The only moment we need a copy is when we actually commit a key. What do you think of having move semantic instead of copy semantic ? That should solve the  `PSA_KEY_USAGE_COPY ` limitation.

Reply:
What exactly do you mean by "move semantic"? I wasn't aware of such concept in the context of PSA. Btw. I checked internally with our crypto teams and according to them `psa_copy_key` should already be fixed on our platforms, so maybe my comment is no longer relevant. I'll try to test your PRs on our platforms tomorrow or on Wednesday and let you know if this actually blocks us or not.

Reply:
By move semantic I was refering to the C++ concepts were object can move handles to unique resources. In our case, the handle is the key ID and we can move the handle from the temporary key to the persistent one. However it creates its own sets of issues because the keys should now be persistent per default.

Thinking a bit more about the issue with `psa_copy_key` is there an issue with Nordic implementation if the copy happens from volatile storage to the persistent/secure element storage ? I was thinking only the copy from secure element could be problematic.

Reply:
Right, so the "move" semantic is more-less equivalent to the current implementation of `PSAOperationalKeystore`.

> Thinking a bit more about the issue with psa_copy_key is there an issue with Nordic implementation if the copy happens from volatile storage to the persistent/secure element storage?

I'll test it now, but I do remember there was some issue with that (but probably more caused by lack of proper driver implementations than by hardware) as I originally implemented `PSAOperationalKeystore` similarly to what you have in your PR, but then I switched to the no-copy approach.

Reply:
OK, I can confirm psa_copy_key already works on our platforms (at least copying volatile keys), so we should be fine with your changes. Please disregard this thread.